app/controllers/api/v2/hosts_controller.rb

```diff
@@ -126,6 +126,30 @@ def status
         render :json => { :status => @host.host_status }.to_json if @host
       end
 
+      api :GET, "/hosts/:id/vm_compute_attributes", N_("Get vm attributes of host")
+      param :id, :identifier_dottable, :required => true
+      description <<-eos
+Return the host's compute attributes that can be used to create a clone of this VM
+      eos
+
+      def vm_compute_attributes
+        if @host
+          attrs = @host.vm_compute_attributes
+          safe_attrs = {}
+          attrs.each_pair do |k,v|
+            # clean up the compute attributes to be suitable for output
+            if v.is_a?(Proc)
+              safe_attrs[k] = v.call()
+            elsif v.respond_to?('parent')
+              # don't add folders, causes recursive json issues
+            else
+              safe_attrs[k] = v
+            end
+          end
+        end
+        render :json => { :compute_attributes => safe_attrs }.to_json if @host
```

Comment:
`.to_json` is unnecessary as `render :json` already *JSONizes* the hash
Isn't the `if @host` redundant (see line 136)?